tutorials/continuous-deployment-pipeline-google-container-engine-with-codeship/index.md

```diff
@@ -0,0 +1,270 @@
+1.  Click `Create Service Account`.
+1.  Enter a name.
+1.  Click `Select a Role` and choose the following permissions:
+
+    * Project &rarr; Service Account Actor
+    * Container &rarr; Container Engine Admin
+    * Storage &rarr; Storage Admin
+1.  Select `Furnish a new private key` and leave the option on `JSON`.
+1.  Click Create, and then close the dialog once it is created.
+
+After the service account is created, a `JSON` file with your credentials
+automatically downloads to your computer. This file will be used in the next
+step.
+
+### Step 2: Setup environment variables
+
+The `hello-express` source code includes the file `example.env`.  Rename this file
+to `.env`.
+
+Inside the file, you will replace the `...` with your Google Cloud Platform
+project details as follows:
+
+1.  `DEFAULT_ZONE`: Select a default [compute zone](https://cloud.google.com/compute/docs/regions-zones/regions-zones#available)
+    like `us-central1-b`
+1.  `APP_NAME`: Give your application a name; in this case use `hello-express`
+1.  `CONTAINER_CLUSTER`: Give your container cluster a name; in this case use
+    `hello-express-cluster`
+1.  `GOOGLE_PROJECT_ID`: The `Project ID` is found in the [Google Cloud Engine dashboard](https://console.cloud.google.com/home/dashboard)
+1.  `GOOGLE_AUTH_EMAIL`: Use the `Service Account ID` for the [service account](https://console.cloud.google.com/iam-admin/serviceaccounts)
+    created in the last step.
+1.  `GOOGLE_AUTH_JSON`: Add the credentials downloaded in the previous step
+    here. You must replace the newlines with spaces by running
+    `tr '\n' ' ' < your_file_name`.
+
+Save the `.env` file once these items are finished.
+
+### Step 3: Run initial deployment
+
+You need to push an image and create your clusters in Container Engine initially before you
+can set up a fully automated pipeline in Codeship. The `hello-express` source code includes a `bin` folder with `bash` scripts that perform these tasks using the
+Codeship Jet CLI.
+
+1.  Make sure the scripts in the `bin` folder are executable. In a terminal
+    window, navigate to the project folder and run the following command:
+
+        chmod -R +x ./bin
+
+1.  Update the `image` line in the `codeship-service.yml` file with your Google
+    Cloud Platform project ID.
+
+        app:
+          build:
+            dockerfile: Dockerfile
+            image: gcr.io/YOUR_PROJECT_IDE/hello-express #update this line using your Google Cloud Platform project ID
+        ...
+
+1.  Update the `image_name` line in the `codeship_steps.yml` file with your
+    Google Cloud Platform project ID.
+
+        - name: build-image
+          service: app
+          command: echo "Build completed"
+        - name: push-image-with-sha
+          service: app
+          type: push
+          image_name: "gcr.io/YOUR_PROJECT_ID/hello-express" #update this line using your Google Cloud Platform project ID
+          image_tag: "{{printf \"%.8s\" .CommitID}}"
```

Comment:
@JimTravis Is our publishing tool going to think this is a Django variable and get all confused?

Reply:
I can put these back in the triple tick markup and label as yaml, if we need to. Let me know!

Reply:
If it's going to be a problem, using triple backticks won't make a difference.

Reply:
Besides, can't use triple backticks within a list using our markdown renderer.

Reply:
I've thought about it some more, and it shouldn't be a problem.

Reply:
Thanks for the clarification. Good to know.